lib/typed_ecto_schema/ecto_type_mapper.ex

```diff
@@ -46,8 +46,13 @@ defmodule TypedEctoSchema.EctoTypeMapper do
     |> add_nil_if_nullable(field_is_nullable?(nullable_default, function_name, opts))
   end
 
-  # Gets the base type for a given Ecto.Type.t()
-  @spec base_type_for(Ecto.Type.t(), field_options()) :: Macro.t()
+  # Gets the base type for a given Ecto.Type.t() or an AST representing a referenced type
+  @spec base_type_for(Ecto.Type.t() | {String.t(), Ecto.Type.t()} | Macro.t(), field_options()) ::
+          Macro.t()
+  defp base_type_for({source, actual_type}, opts) when is_binary(source) do
```

Comment:
Ah, nevermind. Parametrized types in `Ecto.Type.t()` are represented with a triple and not a tuple and we don't even extract it correctly because we just get the second argument from the field macros while parametric types receive the options. Just realized we are just hardcoding our inference for the built-in Enum type.
So ignore this comment hahaha